kebechet/kebechet_runners.py

```diff
@@ -0,0 +1,171 @@
+    slug = slug[1:]
+    namespace = slug.split("/")[0]
+    project = slug.split("/")[1]
+
+    service_url = f"{scheme}://{host}"
+
+    run(
+        service_type=service,
+        service_url=service_url,
+        namespace=namespace,
+        project=project,
+        enabled_managers=["thoth-provenance", "thoth-advise"],
+        analysis_id=analysis_id,
+    )
+
+
+def run(
+    service_type: str,
+    namespace: str,
+    project: str,
+    service_url: Optional[str],
+    enabled_managers: List[str] = list(REGISTERED_MANAGERS.keys()),
+    parsed_payload: Optional[Dict[Any, Any]] = None,
+    metadata: Optional[Dict[str, Any]] = None,
+    analysis_id: Optional[str] = None,
+) -> None:
+    """Run Kebechet using provided YAML configuration file."""
+    token = os.getenv(f"{service_type}_KEBECHET_TOKEN")
+
+    ogr_service = create_ogr_service(
+        service_type=service_type,
+        service_url=service_url,
+        token=token,
+        github_app_id=os.getenv("GITHUB_APP_ID"),
+        github_private_key_path=os.getenv("GITHUB_PRIVATE_KEY_PATH"),
+    )
+
+    with download_kebechet_config(ogr_service, namespace, project) as f:
+        config = _Config.from_file(f)
+
+    slug = f"{namespace}/{project}"
+
+    managers = config.managers
+
+    for manager in managers:
+        # We do pops on dict, which changes it. Let's create a soft duplicate so if a user uses
+        # YAML references, we do not break.
+        manager = dict(manager)
+        try:
+            manager_name = manager.pop("name")
+        except Exception:
+            _LOGGER.exception(
+                "No manager name provided in configuration entry for %r, ignoring entry",
+                slug,
+            )
+            continue
+
+        if manager_name not in enabled_managers:
+            _LOGGER.debug(
+                "Skipping manager %r because it is not in the list of enabled managers.",
+                manager_name,
+            )
+            continue
+
+        kebechet_manager = REGISTERED_MANAGERS.get(manager_name)
+        if not kebechet_manager:
+            _LOGGER.error("Unable to find requested manager %r, skipping", manager_name)
+            continue
+
+        _LOGGER.info("Running manager %r for %r", manager_name, slug)
+        manager_configuration = manager.get("configuration") or {}
+
+        if analysis_id:
+            manager_configuration["analysis_id"] = analysis_id
+
+        try:
+            instance = kebechet_manager(
+                slug=slug,
+                service=ogr_service,
+                service_type=service_type,
+                parsed_payload=parsed_payload,
+                token=token,
+                metadata=metadata,
+            )
+            instance.run(**manager_configuration)
+        except Exception as exc:  # noqa F841
+            _LOGGER.exception(
+                "An error occurred during run of manager %r %r for %r, skipping",
```

Comment:
For future improvements - we should open an issue to the corresponding repo to have this reported to users.